src/host.c

```diff
@@ -626,6 +626,28 @@ cache_store (const char *host, struct address_list *al)
     }
 }
 
+void 
+cache_store (const char *host, const char *ip)
+{
+  if (!is_valid_ipv4_address(ip, ip + strlen(ip))) {
+    logputs(LOG_ALWAYS, "Not a valid IP address: %s", ip);
+    abort();
```

Comment:
This is not a best way to finish execution. I would prefer to see you return an error code or at least `false` boolean as a signal to caller to leave `main()` with not `0` exit code.

Reply:
Ok, makes sense, thanks!